lib/galaxy/tools/toolbox/base.py

```diff
@@ -828,16 +828,19 @@ def quick_load(tool_file, async_load=True):
                 return None
 
         tool_loaded = False
-        for name in os.listdir(directory):
-            if name.startswith('.' or '_'):
-                # Very unlikely that we want to load tools from a hidden or private folder
-                continue
-            child_path = os.path.join(directory, name)
-            if os.path.isdir(child_path) and recursive:
-                self.__watch_directory(child_path, elems, integrated_elems, load_panel_dict, recursive)
-            elif self._looks_like_a_tool(child_path):
-                quick_load(child_path, async_load=False)
-                tool_loaded = True
+        try:
+            for name in os.listdir(directory):
+                if name.startswith('.' or '_'):
```

Comment:
`if name.startswith(('.', '_')):` (tuples work here as of Python 2.5)
Because `'.' or '_'` will evaluate to `'.'`

Reply:
Great catch :)